src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let joinString = '';
+
+    if (this.length !== 0) {
+      for (let element of this) {
+        if (element === undefined || element === null) {
+          element = '';
+        }
+
+        joinString += `${element}${separator}`;
+      }
+    }
+
+    return joinString.slice(0, joinString.lastIndexOf(separator));
```

Comment:
Really clever solution 👍 , but let's try in these tasks not to use another array methods.